app/experimenter/nimbus-ui/src/components/AppLayoutWithSidebar/index.test.tsx

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import { screen, waitFor } from "@testing-library/react";
+import AppLayoutWithSidebar from ".";
+import { renderWithRouter } from "../../lib/helpers";
+import { BASE_PATH } from "../../lib/constants";
+import { RouteComponentProps, Router } from "@reach/router";
+import App from "../App";
+
+describe("PageNew", () => {
+  it("renders app layout content with children", () => {
+    renderWithRouter(
+      <AppLayoutWithSidebar>
+        <p data-testid="test-child">Hello, world!</p>
+      </AppLayoutWithSidebar>,
+    );
+    expect(screen.getByTestId("AppLayoutWithSidebar")).toBeInTheDocument();
+    expect(screen.getByTestId("test-child")).toBeInTheDocument();
+  });
+
+  describe("navigation links", () => {
+    const SidebarRoot = (props: RouteComponentProps) => (
+      <AppLayoutWithSidebar>
+        <p>Hello, world!</p>
+      </AppLayoutWithSidebar>
+    );
+
+    it("renders expected URLs", () => {
+      renderWithRouter(
+        <Router>
+          <SidebarRoot path="/:slug" />
+        </Router>,
+        { route: `/my-special-slug` },
+      );
+      expect(screen.getByTestId("nav-home")).toHaveAttribute("href", BASE_PATH);
+      expect(screen.getByTestId("nav-edit-overview")).toHaveAttribute(
+        "href",
+        `${BASE_PATH}/my-special-slug/edit/overview`,
+      );
+      expect(screen.getByTestId("nav-edit-branches")).toHaveAttribute(
+        "href",
+        `${BASE_PATH}/my-special-slug/edit/branches`,
+      );
+      expect(screen.getByTestId("nav-request-review")).toHaveAttribute(
+        "href",
+        `${BASE_PATH}/my-special-slug/request-review`,
+      );
+    });
+
+    it("renders expected active page class", async () => {
+      function pushState(page: string) {
+        window.history.pushState(
+          {},
+          "",
+          `${BASE_PATH}/my-special-slug/${page}`,
+        );
+      }
+      const {
+        history: { navigate },
+      } = renderWithRouter(<App basepath="/" />, {
+        route: `/my-special-slug/`,
+      });
+
+      pushState("edit/overview");
```

Comment:
This may not be the cleanest way to test this but it was a lot of trial and error to get it working in an at least sort of acceptable way. `location.pathname` is set to `/` in Jest even with page navigations and `pushState` seemed the best way to change it. 

reach-router's `isCurrent` (see on the [`getProps` section of docs](https://reach.tech/router/api/Link)) page for active nav compares `location.pathname` to `href`, but even with them matching, `isCurrent` would always come back as `false`. What I have works locally, in Storybook (after clicking on something in the nav, which is the same with `isCurrent`), and tests now, so I'm inclined to think, "good enough for now"?